frontend/src/pages/view-book.jsx

```diff
@@ -1,27 +1,47 @@
 import React, { useEffect, useState } from 'react';
 import { useParams } from 'react-router-dom';
 import axios from 'axios';
-import { URL_BOOK_SVC } from '../configs.js';
-import Box from '@mui/material/Box';
-import ImageListItem from '@mui/material/ImageListItem';
-import Typography from '@mui/material/Typography';
-import { Button } from '@mui/material';
+import { URL_BOOK_SVC, URL_COLLECTION_SVC } from '../configs.js';
+import {
+  Box,
+  Button,
+  ImageListItem,
+  Typography,
+} from '@mui/material';
 
 const ViewBook = () => {
   const [book, setBook] = useState([]);
+  const [entry, setEntry] = useState({});
+  const [inCollection, setInCollection] = useState(false);
   const { book_id } = useParams();
+  const user_id = localStorage.getItem('userId');
 
-  useEffect(() => {
-    const fetchBook = async (id) => {
-      try {
-        const res = await axios.get(`${URL_BOOK_SVC}/${id}`);
-        setBook(res.data[0]);
-      } catch (err) {
-        console.log(err);
+  const fetchBook = async (id) => {
+    try {
+      const res = await axios.get(`${URL_BOOK_SVC}/${id}`);
+      setBook(res.data[0]);
+    } catch (err) {
+      console.log(err);
+    }
+  }
+
+  const fetchEntry = async (id) => {
+    try {
+      const res = await axios.get(`${URL_COLLECTION_SVC}/${user_id}/${id}`);
+      if (res.data.length !== 0) {
+        setEntry(res.data[0]);
+        setInCollection(true);
       }
+    } catch (err) {
+      console.log(err);
     }
+  }
```

Comment:
may wanna check ; for const declarations in all files

Reply:
same comment as above!